### PR TITLE
Add support for @file args

### LINKED ISSUE #95: Support rust lld @file command line arguments
I don't know exactly why this is a thing, but when the arguments passed to rust lld are too long (?) it gets passed a special argument [(at the bottom of this page)](https://doc.rust-lang.org/rustc/command-line-arguments.html) that refers to a file. Currently, flip-link doesn't seem to handle this.

I've made a fork that "fixes" the bug by just hardcoding support and using a bunch of unwraps [here](https://github.com/great-houk/flip-link/tree/at-file-args), but I imagine someone who knows how linkers work and can make a good solution should be the one to make the PR.

### PR DESCRIPTION
**Pull request by great-houk**
Fixes #95

I couldn't reproduce the original error I encountered, so I couldn't test if this works with rustc generated files, but based on the code for my initial fix, it should work. If someone can find a way to make rustc emit the at-file args so this can be properly tested, that would be great, but I'm very confident in it as it is.

### CODE REVIEW

File: src/argument_parser.rs [hunk 1]
```diff
@@ -22,3 +27,34 @@ pub fn get_search_targets(args: &[String]) -> Vec<Cow<str>> {
...
+        if let Some(arg) = arg.strip_prefix('@') {
+            // The normal linker was able to open the file, so this *should* never panic
+            let file = File::open(arg).expect(&format!(
+                "Unable to open {arg}, this should never happen and should be reported"
+            ));
+            let reader = BufReader::new(file);
+            for line in reader.lines() {
+                // Same as above, normal linker succeeded so we should too
+                let line = line.expect(&format!(
+                    "Invalid file ({arg}), this should never happen and should be reported"
```
Comment (Urhengulas) [inline src/argument_parser.rs:45]: Shouldn't that be?
```suggestion
                    "Invalid file (arg}, this should never happen and should be reported"
```

**Review by Urhengulas — CHANGES_REQUESTED** [review]
Looks good, only one nit

File: src/argument_parser.rs (hunk 1 above)
Comment (great-houk) [reply src/argument_parser.rs:45]: What do you mean? Do you think it should be "{arg}", without the parentheses?

Comment (Urhengulas) [reply src/argument_parser.rs:45]: Yes

**Review by Urhengulas — APPROVED** [review]

**Pull request merged**